lib/interpreters/periodic_visits.rb

```diff
@@ -186,15 +186,24 @@ def generate_services(vrp)
     end
 
     def build_vehicle(vrp, vehicle, vehicle_day_index, rests_durations)
-      new_vehicle = duplicate_safe(
-        vehicle,
-        id: "#{vehicle.id}_#{vehicle_day_index}",
-        global_day_index: vehicle_day_index,
-        skills: associate_skills(vehicle, vehicle_day_index),
-        rests: generate_rests(vehicle, vehicle_day_index, rests_durations),
-        sequence_timewindows: [],
-      )
-      @equivalent_vehicles[vehicle.id] << new_vehicle.id
+      new_vehicle_hash = JSON.parse(vehicle.to_json, symbolize_names: true)
+      new_vehicle_hash.delete(:id)
+      new_vehicle_hash.delete(:start_point_id)
+      new_vehicle_hash.delete(:end_point_id)
```

Comment:
May be we can have something like `except: [:id, :start_point_id, :end_point_id, :sequence_timewindows]` in as_json -- similar to  https://apidock.com/rails/ActiveModel/Serializers/JSON/as_json

I am not sure if this is automatic. Since we are implementing the `as_json` function, I guess we need to implement it via  `options[:except]`. What do you think is it feasible? It would save us from deleting fields manually.

Reply:
@braktar you already did this, right? In  case you kept it alive, I didn't resolve it but if that is not hte case you can resolve it.